### PR TITLE
Added the post comment by article_id feature and completed the error …

### PR DESCRIPTION
**Pull request by ZalwayJH**
…handling

### CODE REVIEW

File: Api/models/articlesModel.js [hunk 1]
```diff
@@ -77,3 +77,26 @@ exports.updateArticleById = (newVotes, id) => {
       return rows[0];
     });
 };
+
+exports.sendComments = (id, newComment) => {
+  const usernameGreenList = ["butter_bridge", "icellusedkars"];
```
Comment (CatHoang) [inline Api/models/articlesModel.js:82]: I would leave this check out as your front end you may want there to be other users unless you make this check dynamic by query the database

File: __tests__/seed.test.js [hunk 2]
```diff
@@ -310,3 +310,45 @@ describe("get/api/articles", () => {
...
+  test("returns an error when the request body endpoints are incorrect", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ AuthorName: "icellusedkars", Pbody: "Good doggo much wow" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
```
Comment (CatHoang) [inline __tests__/seed.test.js:354]: what would happen if there was a valid article but no comments? What would we expect? Also I would also think about error handling an invalid article id e.g /api/articles/apples/comments

File: __tests__/seed.test.js [hunk 3]
```diff
@@ -310,3 +310,45 @@ describe("get/api/articles", () => {
...
+  test("returns an error when the username does not exist", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: "gollum", body: "whats taters precious?" })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("User does not exist");
+      });
+  });
+  test("returns an error when the comment isnt a string", () => {
```
Comment (CatHoang) [inline __tests__/seed.test.js:336]: I would think about how you want to design your backend, if people were to post comments that was just numbers - I would imagine that would be ok. If would be frustrating if I was on reddit and I wanted to comment 10000000 and was not allowed

**Review by CatHoang — COMMENTED** [review]
Hi Joel, some great stuff here. I have left you feedback so just had a read but feel free to merge. I would also delete any files like messingAbout.sql once you are no longer using them as we want a clean project

File: __tests__/seed.test.js (hunk 2 above)
Comment (ZalwayJH) [reply __tests__/seed.test.js:354]: Ah yeah im sorry about this, made some really silly mistakes but thank you for the feedback,  can i just double check does no comments mean an empty string like "" or nothing at all?

Comment (ZalwayJH) [reply __tests__/seed.test.js:354]: oh or do you mean if the query was /ap/1/ ?

**Pull request merged**